packages/transaction-controller/src/types.ts

```diff
@@ -1722,6 +1722,9 @@ export type PublishBatchHookTransaction = {
 
   /** Signed transaction data to publish. */
   signedTx: Hex;
+
+  /** Type of the nested transaction. */
+  type?: TransactionType;
```

Comment:
Why do we need the `type` here?

The intent of providing the `id` is so the hook can require any specific state it needs via the transaction controller state and that reference.